### PR TITLE
Add CasperJS functional test suite

### PR DESCRIPTION
**Pull request by alexgibson**
This is now working locally and also running successfully on Travis while pointed at dev.

I still need to figure out a way to get Travis to start up a local bedrock server to point Casper at, so we can test against changes in PR's instead of what's already on dev.

### CODE REVIEW

**Comment by Lord-of-Darkness** [comment]
Desarrolladores y líderes de Mozilla, que son travestis culo!

Alguien debe colgar ustedes para arriba en sus propias bolas y dejar que su perro muerda su caricatura de una polla!

Posteriormente deben ponerlo tiempo que sea necesario por la garganta hasta que traga a sí mismos en el suicidio!

Ustedes Mozilla está ... Estúpido de mierda coños!
--------------------------------------------------------
Developers and leaders of Mozilla, which are shemales ass!

Someone should hang you up in your own balls and let your dog bite his caricature of a cock!

They should then put it long as needed by the throat until he swallows themselves in suicide!

You Mozilla is... Stupid fucking pussies!

**Comment by alexgibson** [comment]
Ok I'm going to mark this one as ready. Some notes for reviewers:

* I've added documentation on how to set up / run and write tests. Would love feedback here on if things make sense.
* This PR is not a complete suite of tests for the entirety of bedrock. We have tests here for what I think are a good range of pages to get us going, but we can always add more tests after this PR merges and flesh things out.
* This PR does not include running against Travis just yet. I'm enlisting the help of @jgmize to get this working once this is in master.

**Comment by schalkneethling** [comment]
@alexgibson from my experience working with this and contributing to it, I like it a lot! Excellent work! r+

**Comment by schalkneethling** [comment]
@alexgibson We should probably add a note about the Twitter/Blog feed bits, or point to the relevant place in the documentation as that was something that tripped me up in the beginning.

**Comment by alexgibson** [comment]
Thanks @schalkneethling - added a note to the docs about running db sync first before running the tests.

**Comment by alexgibson** [comment]
@jpetto - if you have some available time, would appreciate another pair of eyes to look this over before we merge?

**Comment by jpetto** [comment]
Will look today!

File: docs/testing.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+This is a shortcut for the following command::
+
+    PHANTOMJS_EXECUTABLE=./node_modules/.bin/phantomjs ./node_modules/.bin/casperjs test tests/functional --config=tests/config.json
+
+This will run all test files found in the ``tests/functional`` directory and assumes you
+have bedrock running at ``localhost`` on port ```8000``
+
+To run a single test suite you can tell CasperJS to excecute a specific file e.g.::
+
+    .../casperjs test tests/functional/home.js ...
```
Comment (jpetto) [inline docs/testing.rst]: Would be clearer to include the full command here. :smile:

Comment (alexgibson) [inline docs/testing.rst]: You're not the first to be confused here, so I will take this as a sign that my explanation is lacking :stuck_out_tongue_winking_eye:

**Comment by jpetto** [comment]
Aside from a couple doc clarifications, looks great to me. Nice!

:dancer:

**Pull request merged**